src/runtime/threading_backend.cc

```diff
@@ -92,8 +116,57 @@ class ThreadGroup::Impl {
 #endif
   }
 
+  void InitSortedOrder() {
+    unsigned int threads = threading::MaxConcurrency();
```

Comment:
Some followups: we might need to remodel MaxConcurrency a bit:

Note that the number of workers can indeed be different from the number of cores. The freq-query will always need to query all the CPU info(using hardware concurrency without considering env variable)